src/utils.ts

```diff
@@ -160,34 +172,61 @@ export const copyAll = async (
   )
   let copiedCount = 0
 
-  for (const copyTarget of copyTargets) {
-    const { src, dest, transform, preserveTimestamps, dereference, overwrite } =
-      copyTarget
+  // group copyTargets by dest to avoid race condition in #14
+  const groupedCopyTargets = copyTargets.reduce(
+    (acc, target) => {
+      const { dest } = target
+      const resolvedDest = path.resolve(rootSrc, rootDest, dest)
+      if (!acc[resolvedDest]) {
+        acc[resolvedDest] = []
+      }
```

Comment:
I think we need to cover the following case as well:

- `./foo` -> `./dist/foo` (`./foo` includes `./bar/a.js`)
- `./bar` -> `./dist/foo/bar` (`./bar` includes `./bar/a.js`)

Reply:
Thanks for pointing out, I did not read through the code and was thinking each dest here is already file level, let me dig deeper then

Reply:
I came up with 2 approaches here, and would like to confirm which one sounds more accepatble
1. for each target, resolves folders into file level small sub-targets, each target represents a single file, in this way groups will also be file level, this could ensure the original target order, but requires more IO reads to scan files as well as program complexity
2. sort targets by resolved dest length, shortest first, and iterate over the targets to find out overlapping paths, this does not incur IO but the order in plugin options is not preserved

Reply:
The second approach sounds more simple than the other and I prefer that one, but I wonder if that keep the semantics (the one later in the target array should always overwrite the one first). If it changes the semantics, I think we should go with the first approach. You can use `fastglob(fastglob.escapePath(dir))` to collect the files in the directory.